overlord/hookstate/ctlcmd/ctlcmd.go

```diff
@@ -91,19 +91,44 @@ func addCommand(name, shortHelp, longHelp string, generator func() command) {
 	}
 }
 
+type ForbiddenCommand struct {
+	Uid     uint32
+	Command string
+}
+
+func (f ForbiddenCommand) Error() string {
+	return fmt.Sprintf("cannot use %q with uid %d, try with sudo", f.Command, f.Uid)
+}
+
+func (f *ForbiddenCommand) Execute(args []string) error {
+	return f
+}
+
+func (f *ForbiddenCommand) setStdout(w io.Writer)                 {}
+func (f *ForbiddenCommand) setStderr(w io.Writer)                 {}
+func (f *ForbiddenCommand) setContext(context *hookstate.Context) {}
+func (f *ForbiddenCommand) context() *hookstate.Context {
+	return nil
+}
+
 // Run runs the requested command.
-func Run(context *hookstate.Context, args []string) (stdout, stderr []byte, err error) {
+func Run(uid uint32, context *hookstate.Context, args []string) (stdout, stderr []byte, err error) {
```

Comment:
context should probably remain first.. uid is more closely associated with args (both are from "the other side").

Reply:
Done.